### ISSUE TITLE
Add functionality in df_to_sheet to append data to an existing sheet

### ISSUE THREAD

**Issue by timdmulligan** [aiguofer/gspread-pandas#86]
Hi Diego! Currently in `gspread_pandas.spread.Spread().df_to_sheet()` you are limited to either wiping the sheet or overwriting existing rows using the `replace` parameter. How feasible would it be to add an option to append to an existing sheet instead of just overwriting when `replace = False`? Based on some googling, I think the answer probably lies in [`values_append`](https://docs.gspread.org/en/latest/api.html#gspread.spreadsheet.Spreadsheet.values_append) within gspread.

If you can get me pointed in the right direction I'll see if I can put together a PR.

**Comment by aiguofer** [comment]
Hey! sorry the notifications for this must have slipped through my e-mail. The above workaround def should work well when the data starts in A1 and the column order is the same, but it gets complicated when it doesn't. 

It seems like an `append_to_sheet` method would be a good addition, we just need to make sure that we address the following corner cases:

- does the data fill all available columns/rows or is the worksheet bigger than the data?
- does the data start at A1 or elsewhere?
- are the columns and their order still the same or has it changed?

I imagine the goal of append over replace is simply to reduce the number of API calls and amount of data sent over the wire, which might complicate things because we might still need to fetch data to figure out the answers to the above questions...

**Comment by mrkobk** [comment]
wondering if there is any updated on this? would indeed be ace if the package would be extend with a append_to_sheet for the very reasons you mentioned above. thanks

**Comment by jb261** [comment]
Agreed, append and prepend (for adding rows to table sorted in reverse chronological order) functionality would be very useful.

> I imagine the goal of append over replace is simply to reduce the number of API calls and amount of data sent over the wire, which might complicate things because we might still need to fetch data to figure out the answers to the above questions...

Yes, reading data from spreadsheet into dataframe, updating, then dumping dataframe into worksheet is a clunky solution, and gets worse as the dataset grows in size.

Yes, would probably need to fetch last row of data (for append) or first row of data (for prepend) to figure out where to write, and of course could have optional parameters for user to specify certain characteristics of data to reduce the number of parameters the function needs to determine on its own.

**Comment by pvpscript** [comment]
Was this functionality ever added?

**Comment by Gracecom1** [comment]
Added in [4.0.0](https://pypi.org/project/gspread-pandas/4.0.0/):

```python
spread.df_to_sheet(new_rows, append=True)
```

Rows go below the data already in the sheet. `append` and `replace` together raise, since they mean opposite things.

You pointed at `values_append`, which was the natural read — but appending *positionally* is a quiet footgun. Sheets get columns renamed between exports and reordered by whoever is using them, and a positional append then writes values under the wrong headers with no error anywhere. So `append` resolves your DataFrame's columns against the header row that is already in the sheet, and a column with no counterpart is left blank rather than shifting the row.

Matching is exact first, then difflib similarity — no configuration, no new dependencies. If `GSPREAD_PANDAS_AI_API_KEY` is set it can additionally ask a model about renames that don't look alike (`revenue` → `Sales Total`), but that is off by default and the feature is complete without it. The model is only ever asked for a name-to-name mapping, and any pair naming a column or header that isn't really present is discarded — so a wrong answer costs you a match, never a misplaced value.

Sorry this took as long as it did.
